client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -54,15 +54,23 @@ const PatternAssembler: Step = ( { navigation } ) => {
 		return pageTemplate;
 	};
 
+	let incrementIndex = 0;
 	const addSection = ( pattern: Pattern ) => {
+		incrementIndex++;
 		if ( sectionPosition !== null ) {
 			setSections( [
 				...sections.slice( 0, sectionPosition ),
 				pattern,
```

Comment:
we need to get the key from the previous pattern. Otherwise, it won't have the key 😅